File: DesignPatternsExercises/java/FlyweightDesignPattern/Exercise2/FlightFactory.java
import java.util.HashMap;
import java.util.Map;

public class FlightFactory {
    private static final Map<String, Flight> flightMap = new HashMap<>();

    public Flight getFlight(String flightKey) {
        Flight flight = __________(2)__________.get(flightKey);
        if (flight == null) {
            flight = __________(3)__________(flightKey);
            flightMap.put(flightKey, flight);
        }
        return flight;
    }
}

File: DesignPatternsExercises/ruby/flyweight_design_pattern/flyweight_design_pattern2/solution/flight_factory.rb
require_relative 'flight'

class FlightFactory
  # def __________(3)__________
  def initialize
    # __________(4)__________ = {}
    @flight_map = {}
  end

  def flight(flight_company)
    # return @flight_map[flight_company] if __________(5)__________.key? flight_company
    return @flight_map[flight_company] if @flight_map.key? flight_company

    # flight = __________(6)__________(flight_company)
    flight = Flight.new(flight_company)
    @flight_map[flight_company] = flight
    flight
  end
end
